src/context/sessionContext/index.tsx

```diff
@@ -64,6 +65,8 @@ export interface ISessionProvider {
   session?: Session;
   sessionRequestInProgress?: boolean;
   onError?: (error: Error) => void;
+  restorePreviousSession?: boolean;
```

Comment:
Should this default to `true`?

Reply:
I'm not sure: I'm afraid that defaulting to true for this could result in confusing behaviour, with state loss, redirection and all. It feels less overwhelming to have it opt-in, but I'm open to making true the default too, as I expect most users will expect this to be true.

Reply:
Wasn't the reason for default it to false that if you set it to `true`, you probably also want to add an `onSessionRestore` callback? (In fact, given that this is a React app where (I think?) that is always the case, maybe we should use the presence of that prop as a sign to enable it?)

Reply:
Well technically you could have an app that does not need `onSessionRestore`, if it has no internal state, but I expect this is a niche case. I think setting `restorePreviousSession` to true if `onSessionRestore` is provided also makes sense.

Reply:
Done in 0d303f77472d9d8865cf44fd5ebf6ccdbefaffc8